File: 213-house-robber-ii/house-robber-ii.java
class Solution {
    public int rob(int[] nums) {
      if(nums.length<2){
        return nums[0];
      }
      if(nums.length==2){
        return Math.max(nums[0],nums[1]);
      }
    //  System.out.println(helper(nums,0,nums.length-1));
      return Math.max(helper(nums,0,nums.length-1),helper(nums,1,nums.length));

     
    }
    static int helper(int[] nums,int i,int j){
         int arr[]=new int[nums.length];
      arr[i]=nums[i];
      arr[i+1]=Math.max(nums[i],nums[i+1]);
      for( i=i+2;i<j;i++){
      arr[i]=Math.max(arr[i-2]+nums[i],arr[i-1]);
      }
     System.out.println(Arrays.toString(arr));
      return arr[j-1];
    }
}

File: 213-house-robber-ii/house-robber-ii.cpp
class Solution {
public:
    int check(vector<int> temp){
        int pre=temp[0];
        int pre2=0;
        int n=temp.size();
       
        for(int i=1;i<n;i++){
            int take=temp[i]+pre2;
            int nottake=pre;
            int curr=max(take,nottake);
            pre2=pre;
            pre=curr;
        }
        return pre;
    }
    int rob(vector<int>& nums) {
        vector<int> temp1,temp2;
        int n=nums.size();
            if (n == 0) return 0;
        if (n == 1) return nums[0];
        for(int i=0;i<n;i++){
            if(i!=0) temp1.push_back(nums[i]);
            if(i!=n-1) temp2.push_back(nums[i]);
            
        }
        return max(check(temp1),check(temp2));
    }
};